test/Log/LogServiceTest.php

```diff
@@ -218,77 +217,106 @@ public function testGetDefaultFormatNotConfigured()
 
         $format = $logService->getDefaultFormat();
 
-        $expected = preg_replace('/%runId%/', $logService->getRunId(), LogService::DEFAULT_FORMAT);
-        $expected .= PHP_EOL;
+        $expected = LogService::DEFAULT_FORMAT;
 
         $this->assertEquals($expected, $format);
     }
 
     /**
-     * Format should contain run ID.
+     * Check custom default format can be set.
      */
-    public function testGetDefaultFormatContainsRunId()
+    public function testSetDefaultFormat()
     {
         $logService = $this->getLogService();
 
-        $format = $logService->getDefaultFormat();
+        $logService->setDefaultFormat('%message%');
 
-        $runId = $logService->getRunId();
+        $this->assertEquals('%message%', $logService->getDefaultFormat());
+    }
 
-        $this->assertContains("ID $runId", $format);
+    /**
+     * Test that a unique ID is generated only once.
+     */
+    public function testGetRunId()
+    {
+        $logService = $this->getLogService();
+
+        $id = $logService->getRunId();
+
+        $this->assertNotNull($id);
+        $this->assertInternalType('string', $id);
+        $this->assertEquals($id, $logService->getRunId());
     }
 
     /**
-     * Check custom default format can be set.
+     * Test that a custom unique ID can be set.
      */
-    public function testSetDefaultFormat()
+    public function testSetRunId()
     {
         $logService = $this->getLogService();
 
-        $logService->setDefaultFormat('%message%');
+        $logService->setRunId('customId');
 
-        $this->assertEquals('%message%' . PHP_EOL, $logService->getDefaultFormat());
+        $this->assertEquals('customId', $logService->getRunId());
     }
 
     /**
-     * Check a custom format can use %runId% placeholder.
+     * Test that format's %runId% placeholder is being replaced by runId.
      */
-    public function testGetDefaultFormatCustomFormatContainsRunId()
+    public function testPrepareFormat()
     {
         $logService = $this->getLogService();
 
-        $logService->setDefaultFormat('ID %runId%: %message%');
+        $format = $logService->prepareFormat(self::DEFAULT_FORMAT);
+
         $runId = $logService->getRunId();
 
-        $format = $logService->getDefaultFormat();
+        $expected = preg_replace('/%runId%/', $runId, self::DEFAULT_FORMAT);
+
+        $expected .= PHP_EOL;
 
-        $this->assertContains($runId, $format);
+        $this->assertContains("ID $runId", $format);
+        $this->assertEquals($expected, $format);
     }
 
     /**
-     * Test that a unique ID is generated only once.
+     * Test exception is thrown when format is null.
      */
-    public function testGetRunId()
+    public function testPrepareFormatForNullFormat()
     {
         $logService = $this->getLogService();
 
-        $id = $logService->getRunId();
+        $this->setExpectedException(Exception::class, 'Format must not be null');
```

Comment:
More specific exception class (InvalidArgumentException).

Reply:
I had to run `sudo apt install php-http` in order to not get the error that is showing up here in the test after my latest commit.

Reply:
That is a basic PHP exception and has nothing to do with HTTP. 

https://www.php.net/manual/en/class.invalidargumentexception.php

You are using the wrong exception class.